tests/scripts/check-generated-files.sh

```diff
@@ -11,14 +17,52 @@ fi
 
 check()
 {
-    FILE=$1
-    SCRIPT=$2
+    SCRIPT=$1
+    TO_CHECK=$2
+    PATTERN=""
+
+    if [ -d $TO_CHECK ]; then
+        for FILE in $TO_CHECK/*; do
+            FILES="$FILE $FILES"
+        done
+    else
+        FILES=$TO_CHECK
+    fi
+
+    for FILE in $FILES; do
+        cp $FILE $FILE.bak
+    done
 
-    cp $FILE $FILE.bak
     $SCRIPT
-    diff $FILE $FILE.bak
-    mv $FILE.bak $FILE
+
+    # Compare the script output to the old files and remove backups
+    for FILE in $FILES; do
+        if ! diff $FILE $FILE.bak >/dev/null 2>&1; then
+            echo "'$FILE' was either modified or deleted by '$SCRIPT'"
```

Comment:
If I introduce a new `*.vcxproj` file, I get told it's been modified or deleted, despite it being an addition.

Reply:
> if you manually add a .vcxproj file and then run check-generated-files.sh the script generate_visualc_files.pl will delete the .vcxproj file if it does not correspond to any app, etc so the change will be reported as a "deleted" file.

That's what I did. I created an artificial invented `.vcproj` file and ran the test. In this case, I guess the script caught the change, and the reported error just confused me, probably because I didn't read it properly. The error is 'x was modified or deleted by script Y', and I guess I misread it as 'x was modified or deleted'. Maybe we could write a better error, but it's only a test script. Forget it.